### ISSUE TITLE
Non-Free dependencies

### ISSUE THREAD

**Issue by IzzySoft** [DigitalNutritionalAssessment/gibsonify#79]
v1.4.0 adds MLKit and Firebase, both are non-free components, and also had APK size growing by about 50% (to 32 MB). Would you consider a free/libre build variant coming without those libraries?

For some background: since October last year, your app can be found [in my F-Droid repo](https://apt.izzysoft.de/fdroid/index/apk/org.gibsonify.gibsonify) – which has a size limit of 30 MB per app and only includes FOSS apps. Inclusion criteria are slightly more relaxed than those at [F-Droid.org](https://f-droid.org/) itself, thus allowing for apps which otherwise would just miss inclusion by a small margin. It's currently the largest 3rd-party F-Droid repository, serving more than thousand apps. Your app now "hit the line" due to the combination of size and non-free components – but instead of simply removing it, I'd much more like the idea of "freeing it" so it can stay.

Do we have a chance with that? Depending on what you need MLKit for, maybe there are alternatives. Some hints around that topic can be found e.g. in my [F-Droid snippet](https://gitlab.com/-/snippets/1896503). For now I just added 2 more markers ("anti-features") triggered by MLKit & Firebase: NonFreeNet & Tracking.

**Comment by IzzySoft** [comment]
Not a single word on this? Any opinion, @akabla @fpervaiz? I'd really like to keep your app in my repo, but you don't make it easy for me :wink:

**Comment by adam-sroka** [comment]
Hi @IzzySoft!

First of all I'd like to apologise for the very late reply - I am no longer a maintainer of this project but I can imagine that since this is a university project mostly maintained by students, it can get tricky at some times to devote enough resources to it and keep up with prompt replies.

On a similar note, even though I really appreciate your effort in distributing Gibsonify in your F-Droid repo, I also could imagine that non FOSS components were added by the new team as they are primarily building Gibsonify for a specific use case by a handful of organisations, and only secondly focussing on it being fully FOSS.

That being said, I'll try to reach out to @akabla or @fpervaiz to check what is their opinion on this.

Thanks for your understanding 💙 !

**Comment by IzzySoft** [comment]
Looks like the new team gave up working on this entirely – not only on it remaining FOSS. So if after having added proprietary elements and tracking the app is no longer maintained even (and given the size is already slightly above the per-app limit in my repo), should I rather remove it there altogether?

@akabla @fpervaiz

**Comment by akabla** [comment]
Dear @IzzySoft 
Thanks very much for your interest and feedback. As you noticed, the project development is currently paused. The issue on github is open, and we will look into it when development restarts, which might be in the next academic year. For now, addressing your concerns are therefore not a priority.
Gibsonify is not ready for broad distribution, and we do not have the capacity to support it right now. So there is no need to include it on your F-Droid repo. I would suggest that you kindly remove it from your repository for now.
Many thanks.

**Comment by IzzySoft** [comment]
Thanks for your response, @akabla! And good to know it's on the list and will be addressed one day.

As suggested by you, I've removed the app from my repo now (effective with the next sync around 7 pm UTC). Please give me a ping once I can pick it up again.

**Comment by adam-sroka** [comment]
> Thanks for your response, @akabla! And good to know it's on the list and will be addressed one day.
> 
> As suggested by you, I've removed the app from my repo now (effective with the next sync around 7 pm UTC). Please give me a ping once I can pick it up again.

I'm really sorry for not following up on this as I said I'd do in my previous message. I'm glad that this got resolved for now, and hopefully the next maintainer(s) of this project can pick this up again in the future.

**Comment by IzzySoft** [comment]
Yes, that would be nice! And as I wrote: once that's accomplished, please give me a ping. I'll gladly reestablish the listing then.
